### PR TITLE
Support prev/next button for multiple studies

### PR DESCRIPTION
**Pull request by kalletlak**
# What? Why?
Fix https://github.com/cBioPortal/cbioportal/issues/3639

### CODE REVIEW

File: src/pages/patientView/PatientViewPage.tsx [hunk 1]
```diff
@@ -79,8 +79,19 @@ export default class PatientViewPage extends React.Component<IPatientViewPagePro
                     {
                         patientViewPageStore.setSampleId(query.sampleId as string);
                     }
-                    patientViewPageStore.patientIdsInCohort = ('navCaseIds' in query ? (query.navCaseIds as string).split(",") : []);
 
+                    let patientIdsInCohort = ('navCaseIds' in query ? (query.navCaseIds as string).split(",") : []);
+                    
+                    let multiStudyCohorts: string[] = patientIdsInCohort
+                    .filter(entity=> entity.split(':').length == 2);
```
Comment (alisman) [inline src/pages/patientView/PatientViewPage.tsx]: why not just do entity.contains(":").  no need to split since you're not saving the result

**Comment by alisman** [comment]
Hi guys, 
Adding @ersinciftc and @jji here.  I'm ok with this being a temporary fix but it seems a slippery slope to me where if we don't start using uniqueId for things, our code will get polluted with this kind of thing.  @jj not that heatmap and pathology report just accept patientId, so seems like they could be susceptible to duplicate ID issue.   
To sum up, maybe we should just look at what it would take to start using a uniqueId (handled by backend)

File: src/pages/patientView/PatientViewPage.tsx [hunk 2]
```diff
@@ -79,8 +79,19 @@ export default class PatientViewPage extends React.Component<IPatientViewPagePro
...
+
+                    if(multiStudyCohorts.length > 0) {
+                        patientViewPageStore.patientIdsInCohort = multiStudyCohorts;
+                    } else {
+                        patientViewPageStore.patientIdsInCohort = patientIdsInCohort.map(entityId=>{
```
Comment (alisman) [inline src/pages/patientView/PatientViewPage.tsx]: why not just map patientIdsInCohort and where we don't detect a compound, make it compound (add studyId)?  Seems simpler to me.

File: src/pages/patientView/PatientViewPage.tsx (hunk 1 above)
Comment (alisman) [reply src/pages/patientView/PatientViewPage.tsx]: actually, see comment below. maybe not necessary to determine if we have multiStudyCohort?

**Comment by jjgao** [comment]
@alisman Agreed on support uniqueId in the long run.

For the moment, I am ok with this solution. In fact, `study_id:sample_id` is now pretty standard throughout the interface -- in query page, download tab, and study view page. 

For heatmap and path report, since they are TCGA samples, it's ok just use the sample ID to pull.

**Comment by jjgao** [comment]
It'd be great if we can release this soon.

**Pull request merged**